packages/paragraph/src/withParagraph.ts

```diff
@@ -0,0 +1,23 @@
+import { PlateEditor, Value, isSelectionAtBlockEnd, isSelectionAtBlockStart } from '@udecode/plate-common';
+import { Editor, Path, move, removeNodes } from 'slate';
+
+
+
+
+export const withParagraph = <
+  V extends Value = Value,
+  E extends PlateEditor<V> = PlateEditor<V>
+>(
+  editor: E
+) => {
+  const { deleteForward, deleteBackward } = editor;
+  editor.deleteForward = (unit) => {
+    if (isSelectionAtBlockStart(editor) && isSelectionAtBlockEnd(editor)) {
```

Comment:
We can simplify this to `!isSelectionExpanded(editor) && isBlockAboveEmpty(editor)`